hibernate-core/src/main/java/org/hibernate/engine/spi/CollectionEntry.java

```diff
@@ -375,7 +375,7 @@ public void setRole(String role) {
 	@Override
 	public String toString() {
 		String result = "CollectionEntry" +
-				MessageHelper.collectionInfoString( loadedPersister.getRole(), loadedKey );
+				MessageHelper.collectionInfoString( loadedPersister != null ? loadedPersister.getRole() : null, loadedKey );
```

Comment:
```suggestion
				MessageHelper.collectionInfoString( loadedPersister != null ? loadedPersister.getRole() : role, loadedKey );
```

Reply:
Does that make sense? From what I can tell, if the persister is null, the role is going to be null as well ...

Reply:
It's actually safer to refer to `role` directly instead of `loadedPersister.getRole()`

```suggestion
				MessageHelper.collectionInfoString( role, loadedKey );
```